src/extractor/guidance/mergable_road_detector.cpp

```diff
@@ -0,0 +1,427 @@
+        LengthLimitedCoordinateAccumulator accumulator(
+            coordinate_extractor, node_based_graph, max_length);
+        SelectStraightmostRoadByNameAndOnlyChoice selector(
+            node_based_graph.GetEdgeData(edge_id).name_id, lhs.bearing, false);
+        graph_walker.TraverseRoad(intersection_node, edge_id, accumulator, selector);
+
+        return std::make_pair(accumulator.accumulated_length, accumulator.coordinates);
+    };
+
+    std::vector<util::Coordinate> coordinates_to_the_left, coordinates_to_the_right;
+    double distance_traversed_to_the_left, distance_traversed_to_the_right;
+
+    const double constexpr distance_to_extract = 100;
+
+    std::tie(distance_traversed_to_the_left, coordinates_to_the_left) =
+        getCoordinatesAlongWay(lhs.eid, distance_to_extract);
+
+    // quit early if the road is not very long
+    if (distance_traversed_to_the_left <= 40)
+        return false;
+
+    std::tie(distance_traversed_to_the_right, coordinates_to_the_right) =
+        getCoordinatesAlongWay(rhs.eid, distance_to_extract);
+
+    if (distance_traversed_to_the_right <= 40)
+        return false;
+
+    const auto connect_again = (coordinates_to_the_left.back() == coordinates_to_the_right.back());
+
+    coordinates_to_the_left = coordinate_extractor.SampleCoordinates(
+        std::move(coordinates_to_the_left), distance_to_extract, 5);
+
+    coordinates_to_the_right = coordinate_extractor.SampleCoordinates(
+        std::move(coordinates_to_the_right), distance_to_extract, 5);
+
+    /* extract the number of lanes for a road
+     * restricts a vector to the last two thirds of the length
+     */
+    const auto prune = [](auto &data_vector) {
+        BOOST_ASSERT(data_vector.size() >= 3);
+        data_vector.erase(data_vector.begin(), data_vector.begin() + data_vector.size() / 3);
+    };
+
+    /* if the coordinates meet up again, e.g. due to a split and join, pruning can have a negative
+     * effect. We therefore only prune away the beginning, if the roads don't meet up again as well.
+     */
+    if (!connect_again)
+    {
+        prune(coordinates_to_the_left);
+        prune(coordinates_to_the_right);
+    }
+
+    const auto are_parallel = util::coordinate_calculation::areParallel(coordinates_to_the_left,
+                                                                        coordinates_to_the_right);
+
+    if (!are_parallel)
+        return false;
+
+    // compare reference distance:
+    const auto distance_between_roads = util::coordinate_calculation::findClosestDistance(
+        coordinates_to_the_left[coordinates_to_the_left.size() / 2], coordinates_to_the_right);
+
+    const auto combined_road_width =
+        0.5 *
+        (std::max<int>(
+             1, node_based_graph.GetEdgeData(lhs.eid).road_classification.GetNumberOfLanes()) +
+         std::max<int>(
+             1, node_based_graph.GetEdgeData(rhs.eid).road_classification.GetNumberOfLanes())) *
+        ASSUMED_LANE_WIDTH;
```

Comment:
Can you simplify expressions like this? It's starting to look like Lisp with our narrow column limit :)